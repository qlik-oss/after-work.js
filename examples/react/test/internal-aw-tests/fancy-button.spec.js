import React from "react";
import renderer from "react-test-renderer";

const span = "span";

const [{ default: FancySpan }] = aw.mock(
  [[require.resolve("../../src/button.jsx"), () => () => span]],
  ["../../src/fancy-button"]
);

describe("FancyButton", () => {
  it("renders fancy with button as span or div", () => {
    const tree1 = renderer.create(<FancySpan>span</FancySpan>).toJSON();
    expect(tree1).toMatchSnapshot();
  });

  // Don't copy paste this. It's just to make sure aw.mock works with a file
  it("renders fancy with button mock from file", () => {
    const [{ default: FancyFileMock }] = aw.mock(
      [
        [
          "**/react/src/button.jsx",
          "./examples/react/test/internal-aw-tests/button-mock.js",
        ],
      ],
      ["../../src/fancy-button"]
    );
    const tree1 = renderer
      .create(<FancyFileMock>file mock</FancyFileMock>)
      .toJSON();
    expect(tree1).toMatchSnapshot();
  });
});
