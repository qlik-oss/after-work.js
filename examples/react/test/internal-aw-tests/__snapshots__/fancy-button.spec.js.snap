// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'renders fancy with button as span or div' 1`] = `
<div>
  span
</div>
`;

exports[`'renders fancy with button mock from file' 1`] = `
<div>
  <span>
    file mock
  </span>
</div>
`;
